plugin/src/main/java/io/jenkins/plugins/casc/core/JenkinsConfigurator.java

```diff
@@ -97,9 +104,39 @@ protected Jenkins instance(Mapping mapping, ConfigurationContext context) {
                 .getter(j -> j.proxy)
                 .setter((o, v) -> o.proxy = v));
 
+        attributes.add(new MultivaluedAttribute<Jenkins, NodeMonitor>("nodeMonitors", NodeMonitor.class)
+                .getter(j -> ComputerSet.getMonitors())
+                .setter((jenkins, nodeMonitors) -> {
+                    DescribableList<NodeMonitor, Descriptor<NodeMonitor>> monitors = ComputerSet.getMonitors();
+                    monitors.clear();
+                    monitors.addAll(nodeMonitors);
+                    for (Descriptor<NodeMonitor> d : NodeMonitor.all()) {
+                        if (monitors.get(d) == null) {
+                            NodeMonitor i = createDefaultInstance(d);
+                            if (i != null) {
+                                monitors.add(i);
+                            }
+                        }
+                    }
+                }));
+
         return attributes;
     }
 
+    private static NodeMonitor createDefaultInstance(Descriptor<NodeMonitor> d) {
+        try {
+            NodeMonitor nm = d.clazz.getDeclaredConstructor().newInstance();
+            nm.setIgnored(true);
+            return nm;
+        } catch (NoSuchMethodException
+                | InstantiationException
+                | IllegalAccessException
+                | InvocationTargetException e) {
+            LOGGER.log(Level.SEVERE, "Failed to instantiate " + d.clazz, e);
```

Comment:
does this happen with many implementations? i.e. I'm wondering if it's better to fail fast.

Reply:
This method is copied from ComputerSet, which is private so I can't use it. I could also make it public in https://github.com/jenkinsci/jenkins/pull/8629
I'm not aware of any implementation that is missing it.